sdk/openai/azure-ai-openai-assistants/src/samples/java/com/azure/ai/openai/assistants/RetrievalToolSample.java

```diff
@@ -5,36 +5,100 @@
+
+    public static void main(String[] args) throws InterruptedException {
         String apiKey = Configuration.getGlobalConfiguration().get("NON_AZURE_OPENAI_KEY");
         String deploymentOrModelId = "gpt-4-1106-preview";
-        String fileName = "source.txt";
-
-        Path filePath = Paths.get("src/samples/java/com/azure/ai/openai/resources/" + fileName);
-//        AssistantsClient client = new AssistantsClientBuilder()
-//                .credential(new KeyCredential(apiKey))
-//                .buildClient();
-//
-//        AssistantCreationOptions assistantCreationOptions = new AssistantCreationOptions(deploymentOrModelId)
-//                .setName("Math Tutor")
-//                .setInstructions("You are a personal math tutor. Answer questions briefly, in a sentence or less.");
-//
-//        Assistant assistant = client.createAssistant(assistantCreationOptions);
-//        System.out.printf("Assistant ID = \"%s\" is created at %s.%n", assistant.getId(), assistant.getCreatedAt());
+        String fileName = "retrieval_sample_java_aoai_sdk.txt";
+
+        Path filePath = Paths.get("src/samples/resources/" + fileName);
+        BinaryData fileData = BinaryData.fromFile(filePath);
+        FileDetails fileDetails = new FileDetails(fileData).setFilename(fileName);
+
+        AssistantsClient client = new AssistantsClientBuilder()
+                .credential(new KeyCredential(apiKey))
+                .buildClient();
+
+        // Upload file for assistant
+        OpenAIFile openAIFile = client.uploadFile(new UploadFileRequest(fileDetails, FilePurpose.ASSISTANTS));
+
+
+        // Create assistant passing the file ID
+        Assistant assistant = client.createAssistant(
+            new AssistantCreationOptions(deploymentOrModelId)
+                .setName("Java SDK Retrieval Sample")
+                .setInstructions("You are a helpful assistant that can help fetch data from files you know about.")
+                .setTools(Arrays.asList(new RetrievalToolDefinition()))
+                .setFileIds(Arrays.asList(openAIFile.getId()))
+        );
+
+        // Create a thread with the assistant just created
+        AssistantThread thread = client.createThread(new AssistantThreadCreationOptions());
+
+        // Assign message to thread
+        client.createMessage(
+            thread.getId(),
+            MessageRole.USER,
+            "Can you give me the documented codes for 'banana' and 'orange'?");
+
+        // Pass the message to the assistant and start the run
+        ThreadRun run = client.createRun(thread, assistant);
+
+        // Wait for the run to complete
+        do {
+            Thread.sleep(500);
+            run = client.getRun(thread.getId(), run.getId());
+        } while (run.getStatus() == RunStatus.IN_PROGRESS
+            || run.getStatus() == RunStatus.QUEUED);
+
+        // List messages from the thread
+        var messages = client.listMessages(thread.getId());
```

Comment:
let's not use `var` but the concrete type instead. It would be more straightforward.